app/src/main/java/com/github/andrezimmermann/gluappsample/client/DetailActivity.java

```diff
@@ -0,0 +1,221 @@
+package com.github.andrezimmermann.gluappsample.client;
+
+import android.app.ProgressDialog;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.app.NavUtils;
+import android.support.v7.app.ActionBarActivity;
+import android.view.MenuItem;
+import android.view.MotionEvent;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.ViewFlipper;
+
+import com.github.andrezimmermann.gluappsample.R;
+import com.github.andrezimmermann.gluappsample.client.detail.Presenter;
+import com.github.andrezimmermann.gluappsample.client.detail.RouteDetailView;
+import com.github.andrezimmermann.gluappsample.client.service.BusDetail;
+import com.github.andrezimmermann.gluappsample.client.service.GluApiSingleton;
+import com.github.andrezimmermann.gluappsample.client.widgets.SimpleAdapter;
+import com.github.andrezimmermann.gluappsample.shared.data.BusDeparture;
+import com.github.andrezimmermann.gluappsample.shared.data.BusStop;
+
+import java.util.List;
+
+
+public class DetailActivity extends ActionBarActivity implements RouteDetailView {
+
+    public static final int EMPTY_ROUTE_ID = -1;
+
+    private Presenter presenter;
+    private ProgressDialog progressDialog;
+
+    private int routeId = EMPTY_ROUTE_ID;
+
+
+    private ViewFlipper viewFlipper;
+    private float lastX;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_detail);
+
+        presenter = new Presenter(this, GluApiSingleton.getInstance());
+
+        viewFlipper = (ViewFlipper) findViewById(R.id.detail_flipper);
+
+        Intent intent = getIntent();
+
+        int routeId = intent.getIntExtra(RouteListActivity.ROUTE_ID, 0);
+        String fullDescription = intent.getStringExtra(RouteListActivity.FULL_DESCRIPTION);
+
+        if (shouldFetchDetail(routeId, this.routeId)) {
+            processRouteId(routeId, fullDescription);
+        }
+
+    }
+
+    // Method to handle touch event like left to right swap and right to left swap
```

Comment:
Could you please add a reference from where you got this code that implements the swipe gesture handler.